lib/rubygems/remote_fetcher.rb

```diff
@@ -109,9 +109,12 @@ def api_endpoint(uri)
       uri
     else
       target = res.target.to_s.strip
+      target_host = URI("https://" + target).host
 
-      if URI("http://" + target).host.end_with?(".#{host}")
-        return URI.parse "#{uri.scheme}://#{target}#{uri.path}"
```

Comment:
Note that previously we incorporated the unparsed target content without it being host trimmed.  The this PR simply makes this much more explicit and, to me, it feels safer.